contranomy-sim/tests/Tests/ContranomySim/ReadElf.hs

```diff
@@ -268,8 +269,12 @@ tests = testGroup "Read ELF Tests"
       let (entry, iMem, dMem) = readElf elf
 
       let iDataMap = I.fromList (L.zip [iStart..] (fromIntegral <$> iData))
-      let dDataMap = I.fromList (L.zip [dStart..] (fromIntegral <$> dData))
-                      `I.union` I.fromList (L.zip [bssStart..] (L.replicate (fromIntegral bssLen) 0))
+      let
+        dDataMap = I.unionWithKey (\k _ _ -> error $
+          "Tests.ContranomySim.ReadElf : Overlapping elements in data memory and device tree at address 0x"
```

Comment:
There is no device tree here, this is the .data and .bss sections being placed together.

I don't think this should ever happen though, because the linker should know better than to overlap those

```suggestion
          "Tests.ContranomySim.ReadElf : Overlapping elements in `.data` and `.bss` memory at address 0x"
```

Reply:
good catch! thanks :)